src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -0,0 +1,604 @@
+            comp.VerifyDiagnostics(
+                // (6,11): error CS1503: Argument 1: cannot convert from 'default' to 'int?'
+                //         M(default);
+                Diagnostic(ErrorCode.ERR_BadArgType, "default").WithArguments("1", "default", "int?").WithLocation(6, 11)
+                );
+        }
+
+        [Fact]
+        public void CannotInferTypeArg()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        M(default);
+    }
+    static void M<T>(T x) { }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(
+                // (6,9): error CS0411: The type arguments for method 'C.M<T>(T)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
+                //         M(default);
+                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "M").WithArguments("C.M<T>(T)").WithLocation(6, 9)
+                );
+        }
+
+        [Fact]
+        public void CannotInferTypeArg2()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        M(default, null);
+    }
+    static void M<T>(T x, T y) where T : class { }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(
+                // (6,9): error CS0411: The type arguments for method 'C.M<T>(T, T)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
+                //         M(default, null);
+                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "M").WithArguments("C.M<T>(T, T)").WithLocation(6, 9)
+                );
+        }
+
+        [Fact]
+        public void InvocationOnDefault()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        default.ToString();
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(
+                // (6,16): error CS0023: Operator '.' cannot be applied to operand of type 'default'
+                //         default.ToString();
+                Diagnostic(ErrorCode.ERR_BadUnaryOp, ".").WithArguments(".", "default").WithLocation(6, 16)
+                );
+        }
+
+        [Fact]
+        public void Cast()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        int x = (int)default;
+        System.Console.Write(x);
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "0");
```

Comment:
Good question. I expect it should be `short`. I'll put a PROTOTYPE test to remind to check that once semantic model work is done.